### ISSUE TITLE
error loading static css on ios

### ISSUE THREAD

**Issue by bxrt** [obsidiansystems/obelisk#790]
Following along the steps outlined in https://github.com/obsidiansystems/obelisk#ios I can build, deploy and run the scaffolding app on an iPhone with iOS 13.5.1.
However, the app does not load the static css that is linked. This is evident because the line "Here is a string defined in Common.Api" is not coloured red. Debugging the app while the phone is connected to a Mac shows "An error occurred trying to load the resource." for the css file in Safari's Web Inspector.

Similarly with my own app, an image placed in the static directory and linked in the html is not displayed in the app on iOS.

Xcode: 11.5
MacOS: 10.15.5

relevant output of `xcodebuild -showsdks`

```
2020-07-19 17:16:10.356 xcodebuild[71739:4196397] [MT] DVTSDK: Skipped SDK /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk; its version (10.2) is below required minimum (13.5) for the iphoneos platform.
iOS SDKs:
	iOS 13.5                      	-sdk iphoneos13.5
[...]
```

Noteworthy: the deploy script places `[...]-obelisk.jpg` in directory: `/private/var/containers/Bundle/Application/[BA9B...B8C]/frontend.app/static`, whereas `[...]-main.css` is placed in `/static`
I have no clue if that is supposed to happen or not.

Edit: In "Noteworthy" above I assumed that the deploy script places the files in these directories, based on the error I saw in Safari's Web Inspector. After some thought, I now believe that the error means to say that Safari expects the files in those locations based on the html, but doesn't find the css file there.

**Comment by 3noch** [comment]
Thanks for the detailed report!

**Comment by bxrt** [comment]
This is the generated HTML according to Safari's Web Inspector:
```
<!DOCTYPE html>
<html>
	<head>
		<base href="/">
		<title>Obelisk Minimal Example</title>
		<link href="static/1sjgs70sb2mpmx4k6608n42f0d1y5xxyaf7f4pl5hl961xzjzy7n-main.css" rel="stylesheet" type="text/css">
	</head>
	<body>
		<h1>Welcome to Obelisk!</h1>
		<p>Here is a string defined in Common.Api</p>
		<img src="static/0dspiyay7ldlnxdk7c7q3wjc9b2m4sy348piy8c6qq4dbs00d9x6-obelisk.jpg">
		<div>This string comes from config/common/example</div>
	</body>
</html>

```

Could it be that the `<base href="/">` element roots the relative url for the css in `/static/1sjgs70sb2mpmx4k6608n42f0d1y5xxyaf7f4pl5hl961xzjzy7n-main.css` instead of in `/private/var/containers/Bundle/Application/[BA9B...B8C]/frontend.app/static/1sjgs70sb2mpmx4k6608n42f0d1y5xxyaf7f4pl5hl961xzjzy7n-main.css` where the file actually is?

If so, then I am curious to know why the `<base href="/">` does not affect the src for the image file.

I'd like to do some experiments, but I am too much of a newbie still to know where the `<base href="/">` is generated in the framework.

**Comment by 3noch** [comment]
The base tag comes from https://github.com/obsidiansystems/obelisk/blob/83d20a948e7fc118f071f16c25856a7d4fba8f6a/lib/frontend/src/Obelisk/Frontend.hs#L105

You could `ob thunk unpack .obelisk/impl` in your project and edit it in there to try that.

But yes the fact that the image works is odd.

**Comment by bxrt** [comment]
Commenting
https://github.com/obsidiansystems/obelisk/blob/83d20a948e7fc118f071f16c25856a7d4fba8f6a/lib/frontend/src/Obelisk/Frontend.hs#L216
and thus removing the base element from the html fixes the issue for the iOS app.

I have not looked at how that impacts other targets for the app.

Forgive my ignorance as a newbie, but I assume that there is some magic done with JSaddle to create a WKWebView and then load the HTMLString. As far as I know `WKWebView` and `loadHTMLString` take `baseURL` as an argument on iOS. Maybe a base element in the html conflicts with or supersedes this `baseURL` argument in some cases?
In that case the base element should not be in the html for iOS. Assuming it is needed for other targets, how should it be added target dependently?

**Comment by 3noch** [comment]
For now removing it as a safe stop gap since we're just setting it to `/` anyway. But in the future we need to actually support different base tags and when we do we'll have to figure out how that works with mobile devices since they obviously don't have different "base"s in the same way.

**Comment by bxrt** [comment]
My laptop (2013 Macbook Pro) is running `$(nix-build -A selftest --no-out-link)` in preparation of a pull request. I've been looking at:
```
...
Linking dist/build/obelisk-selftest/obelisk-selftest ...
running tests
Package has no test suites.
haddockPhase
installing
Installing library in /nix/store/pg7j8nv3q06vs1bvk9r1h50adh1zcf41-obelisk-selftest-0.1/lib/ghc-8.6.5/x86_64-osx-ghc-8.6.5/obelisk-selftest-0.1-JzRqBEOjcAZAeslprFZmHg
Installing executable obelisk-selftest in /nix/store/pg7j8nv3q06vs1bvk9r1h50adh1zcf41-obelisk-selftest-0.1/bin
Warning: The directory
/nix/store/pg7j8nv3q06vs1bvk9r1h50adh1zcf41-obelisk-selftest-0.1/bin is not in
the system search path.
Registering library for obelisk-selftest-0.1..
post-installation fixup
strip is /nix/store/adlsv1lf4c9aiqi0cqhvdbs54q70khdv-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/pg7j8nv3q06vs1bvk9r1h50adh1zcf41-obelisk-selftest-0.1/lib  /nix/store/pg7j8nv3q06vs1bvk9r1h50adh1zcf41-obelisk-selftest-0.1/bin
patching script interpreter paths in /nix/store/pg7j8nv3q06vs1bvk9r1h50adh1zcf41-obelisk-selftest-0.1
building '/nix/store/s2ivhpl1px0k5h0gbimw93gzmk38d763-selftest.drv'...
```
for the last 24 hours. How long is the selftest expected to run?

**Comment by 3noch** [comment]
@bxrt Certainly not that long! Do you have caches setup as per the README? I'll try it on my end too.

**Comment by 3noch** [comment]
Selftest passed for me on your commit. That it hangs for you might mean we have a bug in our test. If you have your cache setup correctly then you can try running `$(nix-build -A selftest --no-out-link) -v` to get more output.

**Comment by bxrt** [comment]
Thank you. Selftest passed after I restarted it:
```
Finished in 288.1435 seconds
33 examples, 0 failures
```
I'll delete the "do" at the end of line 214 to satisfy hlint before submitting the pull request.

**Comment by ali-abrar** [comment]
Fixed in #793
